lib/raty.js

```diff
@@ -0,0 +1,780 @@
+  _bindOut() {
+    this.self.addEventListener(
+      'mouseleave',
+      function (evt) {
+        var score = +this.scoreField.value || undefined;
+        this._apply(score);
+        this._target(score, evt);
+        this._resetTitle();
+
+        if (this.opt.mouseout) {
+          this.opt.mouseout.call(this.self, score, evt);
+        }
+      }.bind(this)
+    );
+  }
+
+  // TODO: model spec
+  _bindOutCancel() {
+    var that = this;
+
+    this.cancelButton.addEventListener('mouseleave', function (evt) {
+      var icon = that.opt.cancelOff;
+
+      if (that.opt.starType !== 'img') {
+        icon = that.opt.cancelClass + ' ' + icon;
+      }
+
+      that._setIcon(this, icon);
+
+      if (that.opt.mouseout) {
+        var score = +that.scoreField.value || undefined;
+
+        that.opt.mouseout.call(that.self, score, evt);
+      }
+    });
+  }
+
+  // TODO: model spec
+  _bindOver() {
+    var that = this;
+    var action = that.opt.half ? 'mousemove' : 'mouseover';
+
+    this.stars.forEach((value) => {
+      value.addEventListener(action, function (evt) {
+        var score = that._getScoreByPosition(evt, value);
+        that._fill(score);
+
+        if (that.opt.half) {
+          that._roundStars(score, evt);
+          that._setTitle(score, evt);
+          that.opt.score = score;
+        }
+
+        that._target(score, evt);
+
+        if (that.opt.mouseover) {
+          that.opt.mouseover.call(that.self, score, evt);
+        }
+      });
+    });
+  }
+
+  // TODO: model spec
+  _bindOverCancel() {
+    var that = this;
+
+    this.cancelButton.addEventListener('mouseover', function (evt) {
+      if (that.opt.readOnly) {
+        return;
+      }
+
+      var starOff = that.opt.path + that.opt.starOff;
+      var icon = that.opt.cancelOn;
+
+      if (that.opt.starType === 'img') {
+        that.stars.forEach((value) => {
+          value.src = starOff;
+        });
+      } else {
+        icon = that.opt.cancelClass + ' ' + icon;
+        that.stars.forEach((value) => {
+          value.className = starOff;
+        });
+      }
+
+      that._setIcon(this, icon);
+      that._target(null, evt);
+
+      if (that.opt.mouseover) {
+        that.opt.mouseover.call(that.self, null);
+      }
+    });
+  }
+
+  // TODO: model spec
+  _buildScoreField() {
+    const input = document.createElement('input');
+    input.name = this.opt.scoreName;
+    input.type = 'hidden';
+    this.self.appendChild(input);
+    return input;
+  }
+
+  // TODO: model spec
+  _createCancel() {
+    let icon = this.opt.path + this.opt.cancelOff;
+    let button = document.createElement(this.opt.starType);
+    const attributes = {
+      title: this.opt.cancelHint,
+      class: this.opt.cancelClass,
+    };
+
+    if (this.opt.starType === 'img') {
+      document.width = '20px';
+      document.height = '20px';
+      attributes.src = icon;
+      attributes.alt = 'x';
+    } else {
+      button.style.fontSize = '100%';
```

Comment:
Same thing as [this comment](https://github.com/wbotelhos/raty/pull/240/files#r980098139). I think we should not change the user's decision. They can use CSS for that.